test.py

```diff
@@ -347,181 +441,28 @@ async def reap(done, pending, signaled):
         await TestSuite.artifacts.cleanup_before_exit()
 
     console.print_end_blurb()
+    return total_tests, failed_tests
 
 
-def print_summary(failed_tests: List["Test"], cancelled_tests: int, options: argparse.Namespace) -> None:
+def print_summary(failed_tests: List["Test"], cancelled_tests: int, options: argparse.Namespace, failed_pytest_tests: list[SimpleNamespace], total_tests_pytest: int) -> None:
     rusage = resource.getrusage(resource.RUSAGE_CHILDREN)
     cpu_used = rusage.ru_stime + rusage.ru_utime
     cpu_available = (time.monotonic() - launch_time) * multiprocessing.cpu_count()
     utilization = cpu_used / cpu_available
     print(f"CPU utilization: {utilization*100:.1f}%")
-    if failed_tests:
+    if len(failed_tests) > 0 or len(failed_pytest_tests) > 0:
+        all_fails = [*failed_tests, *failed_pytest_tests]
+        total_tests = TestSuite.test_count() + total_tests_pytest
         print("The following test(s) have failed: {}".format(
-            palette.path(" ".join([t.name for t in failed_tests]))))
+            palette.path(" ".join([t.name for t in all_fails]))))
```

Comment:
Good time to switch to f-string and you don't need square brackets inside `.join()`